autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -297,6 +336,22 @@ def _generate_retrieve_user_reply(
             return False, None
 
     def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str = ""):
+        """Retrieve docs based on the given problem and assign the results to the class property `_results`.
+        In case you want to customize the retrieval process, such as using a different vector db whose APIs are not
+        compatible with chromadb or filter results with metadata, you can override this function. Just keep the current
+        parameters and add your own parameters with default values, and keep the results in below type.
+
+        Type of the results: Dict[str, List[List[Any]]]
+            ids: List[string]
+            documents: List[List[string]]
+            metadatas: Optional[List[List[string]]]
+            distances: Optional[List[List[float]]]
```

Comment:
This part is unclear to me. Why not type the results in the function signature? What are ids, documents, etc.?

Reply:
This function has no returns. ids, documents are the id and content of the each chunk and will be used in function  `_get_context`

> This part is unclear to me. Why not type the results in the function signature? What are ids, documents, etc.?